core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -529,10 +530,17 @@ private[spark] class BlockManager(
 
   private def registerWithExternalShuffleServer(): Unit = {
     logInfo("Registering executor with local external shuffle service.")
+    val shuffleManagerMeta =
+      if (conf.get(config.PUSH_BASED_SHUFFLE_ENABLED)) {
+        s"${shuffleManager.getClass.getName}:" +
+          s"${diskBlockManager.getMergeDirectoryAndAttemptIDJsonString()}}}"
```

Comment:
We shall add comments to `ExecutorShuffleInfo` to explain the `shuffleManagerMeta`.

Reply:
BTW, shall we only add metadata for YARN only?

Reply:
After second thinking, it might be better to keep the consistent behavior under push-based shuffle for all the type of cluster managers.